ppafm/common.py

```diff
@@ -17,56 +22,155 @@
+    """
+
+    PBC: bool = True
+    nPBC: typing.List[int] = [1, 1, 1]
+    gridN: typing.List[int] = [-1, -1, -1]
+    gridO: typing.List[float] = [0.0, 0.0, 0.0]
+    gridA: typing.List[float] = [20.0, 0.0, 0.0]
+    gridB: typing.List[float] = [0.0, 20.0, 0.0]
+    gridC: typing.List[float] = [0.0, 0.0, 20.0]
+    FFgrid0: typing.List[float] = [-1.0, -1.0, -1.0]
+    FFgridA: typing.List[float] = [-1.0, -1.0, -1.0]
+    FFgridB: typing.List[float] = [-1.0, -1.0, -1.0]
+    FFgridC: typing.List[float] = [-1.0, -1.0, -1.0]
+    moleculeShift: typing.List[float] = [0.0, 0.0, 0.0]
+    probeType: str = "O"
+    charge: float = 0.00
+    Apauli: float = 18.0
+    Bpauli: float = 1.0
+    ffModel: str = "LJ"
+    Rcore: float = 0.7
+    r0Probe: typing.List[float] = [0.00, 0.00, 4.00]
+    stiffness: typing.List[float] = [-1.0, -1.0, -1.0]
+    klat: float = 0.5
+    krad: float = 20.00
+    tip: typing.Union[str, typing.List] = "s"
+    sigma: float = 0.7
+    scanStep: typing.List[float] = [0.10, 0.10, 0.10]
+    scanMin: typing.List[float] = [0.0, 0.0, 5.0]
+    scanMax: typing.List[float] = [20.0, 20.0, 8.0]
+    scanTilt: typing.List[float] = [0.0, 0.0, -0.1]
+    tiltedScan: bool = False
+    kCantilever: float = 1800.0
+    f0Cantilever: float = 30300.0
+    Amplitude: float = 1.0
+    plotSliceFrom: int = 16
+    plotSliceTo: int = 22
+    plotSliceBy: int = 1
+    imageInterpolation: str = "bicubic"
+    colorscale: str = "gray"
+    colorscale_kpfm: str = "seismic"
+    ddisp: float = 0.05
+    aMorse: float = -1.6
+    tip_base: typing.List = ["None", 0.00]
+    Rtip: float = 30.0
+    permit: float = 0.00552634959
+    vdWDampKind: int = 2
+    Vbias: float = 0.0
+
+    class Config:
+        arbitrary_types_allowed = True
+        validate_assignment = True
+
+    @classmethod
+    def from_file(cls, file_path: typing.Union[str, Path] = Path("params.ini")):
+        """Load parameters from a file."""
+        file_path = Path(file_path)
+        object = cls()
+        suffix = file_path.suffix
+        try:
+            with open(file_path) as file:
+                if suffix == ".ini":
+                    lines = file.readlines()
+                    object.load_ini(lines)
+                elif suffix == ".yaml":
+                    object.apply_options(yaml.safe_load(file))
+                elif suffix == ".toml":
+                    object.apply_options(toml.load(file))
+                else:
+                    raise ValueError(f"Unsupported file extension: {suffix}")
+        except FileNotFoundError:
+            raise ValueError(f"File {file_path} not found")
+        return object
+
+    def apply_options(self, obj):
+        for key, value in obj.items():
+            print("Applying", key, value)
+            if hasattr(self, key) and value is not None:
+                setattr(self, key, value)
+
+    def dump_parameters(self, file_path: typing.Union[str, Path] = Path("params.ini")):
```

Comment:
Also, we should add a test for these methods. Easiest would be to just create some random parameter object, dump, then load it, and check that the loaded object is identical to the original one.

Reply:
For tests - yes. For getting rid of the "ini" format - it would be nice to have some overlap period when both are supported. Also, this functionality allows to perform the conversion. So users can gradually convert their inputs

Reply:
done in https://github.com/Probe-Particle/ppafm/pull/275/commits/2ba6441df8253fba6b0d931734e0a74f6f311ab9